src/MainWindow.vala

```diff
@@ -1229,8 +1228,28 @@ namespace Terminal {
                 }
                 var text = intext.strip ();
 
+                /* warn on pasting text with `sudo` */
                 if ((text.index_of ("sudo") > -1) && (text.index_of ("\n") != 0)) {
-                    var dialog = new UnsafePasteDialog (this, text);
+                    var dialog = new UnsafePasteDialog (
```

Comment:
This 2 blocks of code (the if...else) are duplicated. I think we could refactor them since the only bit that changes is the string literal.

Reply:
Thanks! I suspected it could be cleaned up but hadn't thought about how. I refactored it now, let me know if that makes more sense.

Reply:
Looking much better, thanks!!